### PR TITLE
Add functionality needed in pittgoogle-user repo

### PR DESCRIPTION
**Pull request by troyraen**
Prompted by mwvgroup/pittgoogle-user#5
Used in mwvgroup/pittgoogle-user#20

Capturing some to-dos:
- [x] alphabetize __init__ imports if possible. else add code comments about why the order is what it is
- [x] move schema registry manifest -> yaml
- [x] add forced photometry to alert.dataframe
- [x] standardize alert.msg.publish_time to a datetime (update from_cloud)
- [ ] ~~publish to TestPyPI and test mwvgroup/pittgoogle-user#20 end to end~~ This doesn't work because then it looks on TestPyPI for other dependencies as well.

## Updated changelog:

(copied from CHANGELOG.md)

### Added

-   `Alert` and `Table` classes.
-   Registry for alert schemas and GCP Project IDs.
-   Alert schemas (Avro) and schema maps (yaml).
-   Exceptions: `BadRequest` and `SchemaNotFoundError`.
-   Types: `PubsubMessageLike` and `Schema`.

### Changed

-   Update PubSub classes.

### Removed

-   v0.1 BigQuery functions.

### CODE REVIEW

File: pittgoogle/bigquery.py [hunk 1]
```diff
@@ -1,692 +1,162 @@
...
+        self.projectid = self._auth.GOOGLE_CLOUD_PROJECT
+
+        return self._auth
+
+    @property
+    def id(self) -> str:
+        """Fully qualified table ID."""
+        # make sure we have a projectid. if it needs to be set, call auth
+        if self.projectid is None:
+            self.auth
```
Comment (troyraen) [inline pittgoogle/bigquery.py:136]: Codacy is complaining about line 136 with "Statement seems to have no effect". It does have an effect, as noted in line 134, but I don't know whether this is an okay workflow by general coding standards. If there's no strong argument against it then I'll leave it, but if there is I can try to figure out how to do this differently.

File: pittgoogle/pubsub.py [hunk 2]
```diff
@@ -146,25 +161,171 @@ class Topic:
...
+        self.projectid = self._auth.GOOGLE_CLOUD_PROJECT
+
+        return self._auth
+
+    @property
+    def path(self) -> str:
+        """Fully qualified path to the topic."""
+        # make sure we have a projectid. if it needs to be set, call auth
+        if self.projectid is None:
+            self.auth
```
Comment (troyraen) [inline pittgoogle/pubsub.py:246]: Same as above, Codacy is complaining. See my comment on bigquery.py file.

File: pittgoogle/bigquery.py (hunk 1 above)
Comment (troyraen) [reply pittgoogle/bigquery.py:136]: If it is okay to leave it, I suppose I should still add some logic to catch an error when the auth cannot be constructed  and give the user a helpful explanation. I can imagine a scenario where they just need to know the `id` and they can supply the `projectid` but don't have credentials for an `Auth` handy.

Comment (wmwv) [reply pittgoogle/bigquery.py:136]: I agree with Codacy's complaint.  It's not obvious that accessing the property calls something.  If it was at least `self.auth()` that would make more sense.

**Comment by troyraen** [comment]
@djperrefort can you take a look at the failing test for Build Documentation? It is complaining about distutils and numpy. The PRs that dependabot has opened (#21 thru #24) appear to have the same failure, so I don't think it's something introduced by this PR.

I did a little bit of Google searching and ended up wondering if we need to restrict to python < 3.12. Seems like we should try updating our explicit dependencies first, but I can't tell exactly what's causing the problem. I tried the more obvious numpy updates, but they didn't work.

**Comment by troyraen** [comment]
Docs build fixed in #28

**Pull request merged**